cmd/catalyst-node/catalyst-node.go

```
@@ -60,21 +73,149 @@ func runClient() error {
+		fmt.Printf("current members in load balancer: %v\n", balancedServers)
+
+		// compare membersMap and balancedServers
+		// del all servers not present in membersMap but present in balancedServers
+		// add all servers not present in balancedServers but present in membersMap
+
+		// note: untested as per MistUtilLoad ports
+		for k := range balancedServers {
+			if _, ok := membersMap[k]; !ok {
+				fmt.Printf("deleting server %s from load balancer\n", k)
+				changeLoadBalancerServers(config.mistLoadBalancerEndpoint, k, "del")
+			}
+		}
+
+		for k := range membersMap {
+			if _, ok := balancedServers[k]; !ok {
+				fmt.Printf("adding server %s to load balancer\n", k)
+				changeLoadBalancerServers(config.mistLoadBalancerEndpoint, k, "add")
+			}
+		}
+
 	}
 
 	return nil
 }
 
+func connectSerfAgent(serfRPCAddress string, serfRPCAuthKey string) (*serfclient.RPCClient, error) {
+	return serfclient.ClientFromConfig(&serfclient.Config{
+		Addr:    serfRPCAddress,
+		AuthKey: serfRPCAuthKey,
+	})
+}
+
+func getSerfMembers(client *serfclient.RPCClient) ([]serfclient.Member, error) {
+	return client.Members()
+}
+
+func changeLoadBalancerServers(endpoint string, server string, action string) {
+	url := endpoint + "?" + action + "server=" + server
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		fmt.Printf("Error creating request: %s", err)
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		fmt.Printf("Error making request: %s", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		b, _ := ioutil.ReadAll(resp.Body)
+		fmt.Printf("Error response from load balancer changing servers: %s\n", string(b))
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Printf("Error reading response: %s", err)
+	}
+	fmt.Println("requested mist to " + action + " server " + server + " to the load balancer")
+	fmt.Println(string(b))
+}
+
+func getMistLoadBalancerServers(endpoint string) map[string]interface{} {
```

should also return an error parameter